### PR TITLE
Vet clinic database: query multiple tables

### PR DESCRIPTION
**Pull request by codedit334**
## Hi Reviewer 👋,

In this PR: 
- Created table `owners`.
- Created table `species`.
- Modified table `animals` to:
  - Remove column species.
  - Add column species_id which is a foreign key referencing species table.
  - Add column owner_id which is a foreign key referencing the owners table.
- Inserted data in the `owners` table.
- Inserted data in the `species` table.
- Modifed my inserted animals so it includes the species_id value:
  - If the name ends in "mon" it will be Digimon.
  - All other animals are Pokemon.
- Modify your inserted animals to include owner information (owner_id):
  - Sam Smith owns Agumon.
  - Jennifer Orwell owns Gabumon and Pikachu.
  - Bob owns Devimon and Plantmon.
  - Melody Pond owns Charmander, Squirtle, and Blossom.
  - Dean Winchester owns Angemon and Boarmon.
- Queries:
  - [What animals belong to Melody Pond?](https://drive.google.com/file/d/1lKE2tsBZuB1NJCOQ1sLIgbhy-C0rqdx8/view?usp=share_link)
  - [List of all animals that are pokemon (their type is Pokemon).](https://drive.google.com/file/d/1pOFcgForoJyeNACXgVw6Sgs3qq0y3IoS/view?usp=share_link)
  - [List all owners and their animals, remember to include those that don't own any animal.](https://drive.google.com/file/d/1oyi1O8e3jgvehan3dDos9LlAb-OGvfZq/view?usp=share_link)
  - [How many animals are there per species?](https://drive.google.com/file/d/1mzXkwiEDuJLLlCu1zEHXioO45iJQWzRD/view?usp=share_link)
  - [List all Digimon owned by Jennifer Orwell.](https://drive.google.com/file/d/1tBtv6MmwAD_WGjZmattR86_RTooAjEwt/view?usp=share_link)
  - [List all animals owned by Dean Winchester that haven't tried to escape.](https://drive.google.com/file/d/1t9rIjptUBTrioeGkc3zsPHqJHWinbFuG/view?usp=share_link)
  - [Who owns the most animals?](https://drive.google.com/file/d/1ovGAsCBds_kxCWXXPCzmspgUY2la2HsL/view?usp=share_link)

# 🤝

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -17,4 +17,60 @@ VALUES ('Charmander', '2020-02-08', -11, false, 0, ''),
...
+
+
+INSERT INTO species (name)
+VALUES ('Pokemon'), ('Digimon');
+
+
+ALTER TABLE animals
+SET species_id = 2, owner_id = 1
+WHERE name = 'Agumon';
+
```
Comment (christianbiring1) [inline data.sql]: - [ ] As you may notice in your attached picture, how all the animal's name ending with `mon` doesn't have `Digimon` as a `species_id`, and all other animals have Pokemon instead, Kindly make sure you fix that.

Hint: To achieve that you may consider saying:
```sql
UPDATE animals
   SET species_id = (SELECT id from species WHERE name = 'Digimon') -- Or you can hardcode the species_id here which I wouldn't recommend --
   WHERE name like '%mon';

   UPDATE animals
   SET species_id = (SELECT id from species WHERE name = 'Pokemon')
   WHERE species_id IS NULL;
```

Comment (christianbiring1) [reply data.sql]: - [ ] This being said, you may consider assigning each owner to their animals like so:
```sql
UPDATE animals
 SET owner_id = (SELECT id from owners WHERE full_name = 'Sam Smith')
   WHERE name = 'Agumon';
-- With more animals  you may say for example--

UPDATE animals
 SET owner_id = (SELECT id from owners WHERE full_name = 'Melody Pond')
  WHERE name = 'Charmander' OR name = 'Squirtle' OR name = 'Blossom';
```

File: queries.sql [hunk 2]
```diff
@@ -61,4 +61,42 @@ SELECT neutered, AVG(escape_attempts) as avg_escapes FROM animals GROUP BY neute
...
+-- Project 3
+
+SELECT name FROM animals
+JOIN owners ON owners.id = animals.owner_id
+WHERE owners.full_name = 'Melody Pond';
+
+SELECT animals.name
+FROM animals
+JOIN species ON animals.species_id = species.id
+WHERE species.name = 'Pokemon';
```
Comment (christianbiring1) [inline queries.sql:75]: - [ ] Pokemon is supposed to be the species only for animals that doesn't end with `mon`. once you apply the changes suggested early, you will be good to go. the query here is correct :+1:

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @codedit334 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# STATUS: REQUEST CHANGES :recycle: 


## Highlights
- [ ] Correct queries to answer the questions :heavy_check_mark: 
- [ ] Descriptive PR `title` and `summary` :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @codedit334, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requesteed Changes :heavy_check_mark: 


## Optional suggestions
- [ ] Going forward for the `DATABASE` projects please add screenshots of your queries to the PR description
image

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**